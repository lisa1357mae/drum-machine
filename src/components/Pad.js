import React, { Component } from 'react';

class Pad extends Component {

    render() {
      const { name } = this.props;
      let lightOnStyle, lightOffStyle;
      switch (name) {
        case 'HiHat':
          lightOnStyle = 'hh-circle-active';
          lightOffStyle = 'hh-circle';
          break;
        case 'Snare':
          lightOnStyle = 'sd-circle-active';
          lightOffStyle = 'sd-circle';
          break;
        case 'Kick':
          lightOnStyle = 'k-circle-active';
          lightOffStyle = 'k-circle';
          break;
        default:
          console.error('wrong instrument');
      };
      return (
        <div><div className={this.props.lightOn ? lightOnStyle : lightOffStyle}><p className='pad'>{name}</p></div></div>
      );
    }
}

export default Pad;