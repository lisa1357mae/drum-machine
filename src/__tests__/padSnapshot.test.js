import React from 'react';
import renderer from 'react-test-renderer';
import Pad from '../components/Pad';

test('renders correctly', () => {
  const tree = renderer.create(<Pad />).toJSON();
  expect(tree).toMatchSnapshot();
});
