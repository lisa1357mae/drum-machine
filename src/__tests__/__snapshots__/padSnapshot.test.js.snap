exports[`test renders correctly 1`] = `
<div>
  <div
    className={undefined}>
    <p
      className="pad" />
  </div>
</div>
`;
